eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -11,11 +11,20 @@ parameters:
   - name: SkipCheckoutNone
     type: boolean
     default: false
+  - name: TokenToUseForAuth
+    type: string
+    default: ''
 
 steps:
   - ${{ if not(parameters.SkipCheckoutNone) }}:
       - checkout: none
 
+  - ${{ if ne(parameters.TokenToUseForAuth, '') }}:
+    - pwsh: |
+        $base64Token = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes("nobody:${{ parameters.TokenToUseForAuth }}"))
+        git config set --global "http.extraheader" "AUTHORIZATION: basic $base64Token"
```

Comment:
Should we mark this value as secret via devops logging just in case?

Reply:
Yes, we should mark the base64Token as a secret as well.